### PR TITLE
Check for unauthorized

### LINKED ISSUE #24: setAPIUrlStyle defaults to old api style
UDM Firmware: 1.8.6
Controller Version: 6.0.43

Currently failing `terraform plan` using provider 0.19.2 with `401 Unauthorized`.

`setAPIUrlStyle` sets the new api style only for 200 responses. The api style check will never set the new api paths for controller versions 6.x.x as the code is effectively unreachable. See below...

``` go
	if resp.StatusCode == http.StatusOk {
		// the new API returns a 200 for a / request
		c.apiPath = apiPathNew
		c.loginPath = loginPathNew
		c.statusPath = statusPathNew
		return nil
	}
```

### PR DESCRIPTION
**Pull request by maowenbrad**
Set new api path when targeting UniFi Controller version 6.x.x 
Fixes #24

### CODE REVIEW

**Comment by paultyng** [comment]
I'm guessing you are on a UDM-Pro (not a regular UDM)?

Sorry you hit this hiccup, hopefully you can version pin to workaround while we figure this out. I wonder if we should just check for both responses to handle different versions people may be running as presumably this works for some folks? I'm not sure, its hard to pin down the exact controller version differences since its not a well documented API.

I do not have a UDM-P to test on at home, just a UDM, but the rate at which I get UMD-P bug reports is seriously making me reconsider this 😄

**Comment by paultyng** [comment]
The automated testing on the provider tests against a v5 and v6 docker version of the container, so this seems to be UDM-P specific at least (you can see the nightly test result here which was against 6.0.45: https://github.com/paultyng/terraform-provider-unifi/runs/1814868401?check_suite_focus=true).

**Comment by maowenbrad** [comment]
No need to apologize. This is great work. I really appreciate your work and time. I agree that it is difficult to work with this undocumented API. 

As to the bug at hand, I am using a UDM. Not a UDMP. I have version pinned and that works.

I am new to Golang but I'm willing to work on a more elegant change.

**Comment by paultyng** [comment]
Oh interesting, I have a UDM at home but I guess i'm not offered this firmware upgrade yet, need to dig in to why so then I can maybe poke around here.

**Comment by joshuaspence** [comment]
Unrelated to this PR but we should add test coverage to this code path.

File: unifi/unifi.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (c *Client) setAPIUrlStyle(ctx context.Context) error {
 	defer resp.Body.Close()
 	_, _ = io.Copy(ioutil.Discard, resp.Body)
 
-	if resp.StatusCode == http.StatusOK {
```
Comment (joshuaspence) [inline unifi/unifi.go:115]: FWIW, our current implementation is consistent with another Unifi Go library: https://github.com/unpoller/unifi/blob/master/unifi.go#L193

**Comment by joshuaspence** [comment]
I am guessing this isn't relevant anymore

**Pull request closed without merging**